### PR TITLE
Add Some Helpful Fields

### LINKED ISSUE #2497: openstack flavor show [flavorID] properties
Hello, I use the command `openstack flavor show [flavorID]` ，flavor detail has the properties field，but I'm not found in the https://github.com/gophercloud/gophercloud/blob/0226ea5963656bcd435ab17af475e2345e523fcb/openstack/compute/v2/flavors/results.go#L51

### PR DESCRIPTION
**Pull request by spjmurray**
Add the "extra_specs" field in compute flavors, that can avoid another lookup.  As documented it's for microversions 2.61 and greater. Documented here:

* https://docs.openstack.org/api-ref/compute/?expanded=list-server-groups-detail,create-server-group-detail,show-flavor-details-detail,list-flavors-with-details-detail#list-flavors
* https://docs.openstack.org/api-ref/compute/?expanded=list-server-groups-detail,create-server-group-detail,show-flavor-details-detail#list-flavors

Prior to starting a PR, please make sure you have read our
[contributor tutorial](https://github.com/gophercloud/gophercloud/tree/master/docs/contributor-tutorial).

Prior to a PR being reviewed, there needs to be a Github issue that the PR
addresses. Replace the brackets and text below with that issue number.

Fixes #2497

Links to the line numbers/files in the OpenStack source code that support the
code in this PR:

See commit message... seems I jumped the gun 😸

### CODE REVIEW

File: openstack/networking/v2/networks/results.go
Comment (kayrus) [inline openstack/networking/v2/networks/results.go:99]: it's already implemented. you need to use https://pkg.go.dev/github.com/gophercloud/gophercloud/openstack/networking/v2/extensions/external

File: openstack/compute/v2/flavors/results.go
Comment (kayrus) [inline openstack/compute/v2/flavors/results.go:88]: this is already implemented you need to use https://pkg.go.dev/github.com/gophercloud/gophercloud/openstack/compute/v2/flavors#ListExtraSpecs

**Review by kayrus — COMMENTED** [review]
Looks like everything is already implemented. You just need to use some extra code to obtain the desired information.

File: openstack/compute/v2/flavors/results.go
Comment (spjmurray) [reply openstack/compute/v2/flavors/results.go:88]: It is already implemented, but you can avoid the N extra round trips and just get the data directly in the /details call.  So above and beyond the obvious performance improvement, and load reduction, it cleans up client code a lot, especially if you have to mess about with go routines to hide the extra calls.

File: openstack/networking/v2/networks/results.go
Comment (spjmurray) [reply openstack/networking/v2/networks/results.go]: Yeah, just discovered this via the CI failure!  Can confirm it does work with my app.

File: openstack/compute/v2/flavors/results.go
Comment (spjmurray) [reply openstack/compute/v2/flavors/results.go:88]: To prove my point of how bad it can be here's the naive version:
extra_specs_naive

And when you add in a heap of code to parallelize everything:
extra_specs_parallel

Or, what this proposes, is neither of those things and just getting it in the original call.

Comment (kayrus) [reply openstack/compute/v2/flavors/results.go:88]: fair point :)

**Review by kayrus — APPROVED** [review]

**Comment by spjmurray** [comment]
@kayrus I do note this is a dupe of https://github.com/gophercloud/gophercloud/pull/2561, my bad for not looking hard enough, sorry.  But it has been sat about since February, so I'm unsure if you want to favor one over the other.  Your call.

**Comment by kayrus** [comment]
@spjmurray I also missed this PR, but I'm not a maintainer. @mandre @pierreprinetti have all the cards

**Comment by spjmurray** [comment]
No wonder they always in at poker!

**Comment by pierreprinetti** [comment]
@mandre has reviewed #2561 and I'll let him merge one of the two PRs next week when he's more likely to have time for Gophercloud. Thank you for your ping on this issue, and sorry that it's taking longer than necessary; we're really a bit busy with other projects these days.

**Comment by spjmurray** [comment]
Thanks for your eyes, I'm not surprised with the time, I've got my fingers in CAPO, openstack-cloud-controller-manager, argocd, and my own platform to boot.  Hard to juggle everything at once!

**Comment by kayrus** [comment]
@pierreprinetti can I volunteer to join the gophercloud maintainers list?

**Comment by mandre** [comment]
I'm closing this PR as it's indeed duplicating https://github.com/gophercloud/gophercloud/pull/2561.

**Comment by spjmurray** [comment]
Sounds reasonable, can't expect everyone to want to use DeepEqual everywhere.  I will await its merger!

**Pull request closed without merging**
